sdk/protocol-plugins/src/aave-v3/AAVEv3ProtocolPlugin.ts

```diff
@@ -7,34 +7,66 @@ import {
   Price,
   CurrencySymbol,
   RiskRatio,
+  ChainFamilyName,
+  valuesOfChainFamilyMap,
+  ChainId,
 } from '@summerfi/sdk-common/common'
 import type { AaveV3PoolId } from '@summerfi/sdk-common/protocols'
 import { PoolType, ProtocolName, EmodeType } from '@summerfi/sdk-common/protocols'
 import { BigNumber } from 'bignumber.js'
 import { z } from 'zod'
+import { BaseProtocolPlugin } from '../implementation/BaseProtocolPlugin'
+import { AaveV3LendingPool, AaveV3PoolCollateralConfig, AaveV3PoolDebtConfig } from './Types'
 import {
-  AaveV3LendingPool,
-  AaveV3PoolCollateralConfig,
-  AaveV3PoolDebtConfig,
-  ChainId,
-  IPositionId,
-  ProtocolManagerContext,
-  ProtocolPlugin,
-} from '../interfaces'
-import { AaveV3LikePluginBuilder, filterAssetsListByEMode } from './AAVEv3LikeBuilder'
-import { UNCAPPED_SUPPLY, PRECISION_BI } from './constants'
-
-export class AaveV3Plugin implements ProtocolPlugin<AaveV3PoolId> {
-  public readonly protocol = ProtocolName.AAVEv3
-  supportedChains = [ChainId.Mainnet, ChainId.Arbitrum, ChainId.Optimism]
-  async getPool(aaveV3PoolId: unknown, ctx: ProtocolManagerContext): Promise<AaveV3LendingPool> {
+  AaveV3LikePluginBuilder,
+  filterAssetsListByEMode,
+} from '../implementation/AAVEv3LikeBuilder'
+import { UNCAPPED_SUPPLY, PRECISION_BI } from '../implementation/constants'
+
+export class AaveV3ProtocolPlugin extends BaseProtocolPlugin<AaveV3PoolId> {
+  public static protocol: ProtocolName.AAVEv3 = ProtocolName.AAVEv3
+  public static supportedChains = valuesOfChainFamilyMap([
```

Comment:
I wonder what is the value of this. Shouldn't this be data driven and just validated against the dependency addresses for each chain? In the sense that if we cannot find the protocol entry in the contract addresses for the chain, then we return an error. I see it is only used to check if the input ChainId is in the list of supported chains, but then it is not used anymore in the rest of the plugin 🤔

Reply:
Probably. When we get to that I'll make the change. Think it's just not for this PR